aptos-move/framework/aptos-framework/sources/storage_gas.move

```diff
@@ -209,32 +233,45 @@ module aptos_framework::storage_gas {
         // Check the corner case that current_usage_bps drops before the first point.
         let (left, right) = if (num_points == 0) {
             (&Point {x: 0, y: 0}, &Point {x: BASIS_POINT_DENOMINATION, y: BASIS_POINT_DENOMINATION})
+        } else if (current_usage_bps < vector::borrow(points, 0).x) {
+            (&Point { x: 0, y: 0 }, vector::borrow(points, 0))
+        } else if (vector::borrow(points, num_points - 1).x <= current_usage_bps) {
+            (vector::borrow(points, num_points-1), &Point { x: BASIS_POINT_DENOMINATION, y: BASIS_POINT_DENOMINATION })
         } else {
-            let (i, j) = (0, num_points - 1);
-            while (i < j) {
+            let (i, j) = (0, num_points - 2);
+            while ({
+                spec {
+                    invariant i <= j;
+                    invariant j < num_points;
```

Comment:
`j < num_points - 1` to make line 247 valid?

Reply:
Good catch! `j < num_points` works here, but `j < num_points - 1` seems to be more precise. Done fixing it.